backend/src/test/java/com/google/models/AnomalyTest.java

```diff
@@ -0,0 +1,145 @@
+package com.google.models;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+/** Contain tests for methods in {@link Anomaly} class. */
+@RunWith(JUnit4.class)
+public final class AnomalyTest {
+  private static final int TIMESTAMP_CONSTANT = 1;
+  private static final String METRIC_NAME = "Sample metric name";
+  private static final String DIMENSION_NAME = "Sample dimension name";
+  private static final Map<Timestamp, MetricValue> DATA_POINTS = ImmutableMap.of( 
+      Timestamp.getDummyTimestamp(1), new MetricValue(1), 
+      Timestamp.getDummyTimestamp(2), new MetricValue(2), 
+      Timestamp.getDummyTimestamp(3), new MetricValue(3));
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private static final Anomaly anomaly = new Anomaly(Timestamp.getDummyTimestamp(TIMESTAMP_CONSTANT), 
```

Comment:
move this into the setUp method? that is how it is set up in the AlertTest.java file I think

Reply:
i didn't put this in the `setUp()` method because here `anomaly` is declared with `static final` since it is never changed throughout any of the following unit tests.
in `AlertTest.java`, `alert` is initialized in the `setUp()` method because it is not declared as `static final` as one of the unit tests alters a property of the `alert` object. 

is this okay with you? if not we can further discuss it and i can move it to the `setUp()` method :)